### PR TITLE
Add indexes for commonly used queries

### PR DESCRIPTION
**Pull request by ehlertjd**
These were low-hanging fruit discovered via mongo's [notablescan](https://docs.mongodb.com/manual/reference/parameters/#param.notablescan) parameter, and accessing commonly used endpoints. 

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md
- Extra care is taken and log messages are created for [Critical Data](https://github.com/flywheel-io/core/wiki/Critical-Data-Flows)

### CODE REVIEW

File: tests/integration_tests/python/test_uploads.py [hunk 1]
```diff
@@ -795,9 +795,12 @@ def test_label_upload(data_builder, file_form, as_admin):
...
 
     session = get_full_container(as_admin, '/projects/' + project + '/sessions', 1)
-    session_id = session['_id']
-
     session2 = get_full_container(as_admin, '/projects/' + project + '/sessions', 0)
+
+    # Swap so that session represents test_session1
+    if session['label'] == 'test_session2_label':
+        session, session2 = session2, session
+    session_id = session['_id']
```
Comment (nagem) [inline tests/integration_tests/python/test_uploads.py]: Is this relevant to the indexes being adjusted?

**Comment by nagem** [comment]
Did you do any investigation into how much this affects our index size? [Mongo suggests you create indexes that fit in RAM](https://docs.mongodb.com/manual/tutorial/ensure-indexes-fit-ram/), but I'm not sure what our state before these changes were either.

File: tests/integration_tests/python/test_uploads.py (hunk 1 above)
Comment (ehlertjd) [reply tests/integration_tests/python/test_uploads.py]: Seems to be - we were assuming a sort order, and apparently adjusting indexes caused that assumption to be invalidated.

**Comment by ehlertjd** [comment]
@nagem I've been exploring that a bit with scalability testing - my hope is to end up with some basic RAM guidelines based on collection sizes. (e.g. number of acquisitions or number of sessions).

**Comment by ehlertjd** [comment]
On the dev system, adding indexes took about 7 seconds, and roughly doubled the index sizes for each collection. As discussed, I'll add the background option to index creation, and test before merging.

```
5,808 Acquisitions
31,820 Jobs

=== Before ===
db.acquisitions.totalIndexSize()
1273856 (~1.2 MB)

db.jobs.totalIndexSize()
2469888 (~2.4 MB)


=== After ===
db.acquisitions.totalIndexSize()
1556480 (~1.5 MB)

db.jobs.totalIndexSize()
3059712 (~3.0 MB)
```
